Fix UNION regression

wboettge opened the pull request:
### Description

Previous work to fix ORDER BY clauses with UNION statements introduced a regression. These changes temporarily replace the top-level target list, which is checked to ensure the length does not change. The problem arises when the new target list contains resjunk columns, which results in a different length than the original target list. This change fixes this issue.

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

wboettge committed:
Fix UNION regression

Previous work to fix ORDER BY clauses with UNION statements introduced a
regression. These changes temporarily replace the top-level target list,
which is checked to ensure the length does not change. The problem
arises when the new target list contains resjunk columns, which results
in a different length than the original target list. This change fixes
this issue.

Signed-off-by: Walt Boettge <wboettge@amazon.com>

forestkeeper commented on test/JDBC/target/surefire/surefire1452586778093699594tmp:
```diff
@@ -0,0 +1,55 @@
+classPathUrl.6=/home/wboettge/.m2/repository/org/junit/jupiter/junit-jupiter-engine/5.7.0/junit-jupiter-engine-5.7.0.jar
+classPathUrl.7=/home/wboettge/.m2/repository/org/junit/platform/junit-platform-engine/1.7.0/junit-platform-engine-1.7.0.jar
+classPathUrl.8=/home/wboettge/.m2/repository/org/junit/jupiter/junit-jupiter-params/5.5.2/junit-jupiter-params-5.5.2.jar
+classPathUrl.9=/home/wboettge/.m2/repository/org/apiguardian/apiguardian-api/1.1.0/apiguardian-api-1.1.0.jar
```
that's ignore file for github

wboettge committed:
Re-add cleanup hook and add additional tests

post_transform_sort_clause resets the targetlist back to its original
state. This ends up being necessary in when using UNION ALL, and when
target entries reference the second or deeper joined table. This commit
adds the hook back, and adds tests for these cases.

Signed-off-by: Walt Boettge <wboettge@amazon.com>
